### ISSUE TITLE
Proposal: Infer "Run All"

### ISSUE THREAD

**Issue by ghickman** [opensafely/job-server#309]
We currently special case the `run_all` action in both the job-runner and job-server projects.  In job-server this has been confusing to explain, as seen in #276.

In job-server we special-case it in a fairly simple way, if it's not already in the list extracted from a `project.yaml`, we add it. This works well.  job-runner looks for that action and pulls all the actions from the `project.yaml` it loaded.  However, explaining why one item in the list is different is much more complex, and that's the crux of the issue: "one of these is not like the others".

So, I propose we change things around a little:

* Stop adding `run_all` to the Actions list
* Add a `Select All" button above the Actions list which does exactly what it says on the tin, selects all the Actions in the list
* Save the full list of actions to the `requested_actions` field

Now "Run All" is a shortcut for "click each action please and then force run them".  The difference between `run_all` with or without `force_run_dependencies` is the same as it is with any other action because we've removed the special-casing.

Note: JobRequests save the SHA they were working from so job-runner should always be able to see the same list of actions.  If it can't we have bigger problems.

Thoughts @sebbacon & @evansd?

**Comment by sebbacon** [comment]
Yes, I think explaining "run all" in terms of a UI which ticks things for people is probably the right direction of travel.

How is this the same or different to https://github.com/opensafely/job-server/issues/133?  For example, if I understand correctly, this proposal doesn't allow for "run everything that's not been run before"? 

As a general point on the "expose run-all as a select-all UI concept" which is very hard to solve: an important difference between "select all" and "run all" is that the definition of "all" may change between the user making the jobrequest and the server executing it.  We should probably fix this by ensuring the job-runner executes things at the SHA that was current when the jobrequest was made (or does it already and I missed that?!)

**Comment by evansd** [comment]
> We should probably fix this by ensuring the job-runner executes things at the SHA that was current when the jobrequest was made (or does it already and I missed that?!)

This already happened and you missed it :smile:

There is still a semantic difference between @ghickman's proposal and the current status quo which is that explicitly selecting actions in the UI implies a "force run" that is not implied by using run_all.

I don't have time right this instant to think about alternatives, but just wanted to flag up this difference now.
